### PR TITLE
Alex Shaw

### PR DESCRIPTION
**Pull request by alexjshaw**
Missing a big chunk of work that I did yesterday afternoon but hadn't committed yet!

### CODE REVIEW

File: src/CohortManager.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+          cohort: cohort.cohortName
+        }
+        return studentSearchResult
+      }
+    }
+    throw new Error('No student matching that ID was found')
+  }
+}
+
+module.exports = CohortManager
```
Comment (desokan) [inline src/CohortManager.js:141]: I don't have any addition or subtraction to suggest. Well done with your code outlay.

File: src/CohortManager.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    )
+    if (studentFound) {
+      const deletionIndex = cohortFound.cohortStudents.indexOf(studentFound)
+      cohortFound.cohortStudents.splice(deletionIndex, 1)
+    } else {
+      throw new Error('Student not found')
+    }
+  }
+
+  findStudentByID(studentID) {
```
Comment (procrastian) [inline src/CohortManager.js:121]: Really clear naming conventions and layout of the methods - nice and easy to follow! Good job!

File: src/CohortManager.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+      )
+      if (studentFound) {
+        const studentSearchResult = {
+          ...studentFound,
+          cohort: cohort.cohortName
+        }
+        return studentSearchResult
+      }
+    }
+    throw new Error('No student matching that ID was found')
```
Comment (procrastian) [inline src/CohortManager.js:137]: Does throwing the error in this way give the error message if that test fails? Does this work in the same way as returning a "fail condition" in your function would?

File: src/Student.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class Student {
+  constructor(ID, firstName, lastName, gitName, email) {
+    this.studentID = ID
+    this.firstName = firstName
+    this.lastName = lastName
+    this.gitName = gitName
+    this.email = email
```
Comment (procrastian) [inline src/Student.js:7]: Awesome work! Layout of the files and code is very clean!

File: src/CohortManager.js (hunk 3 above)
Comment (alexjshaw) [reply src/CohortManager.js:137]: Throw new Error will display the error message in the console, so if you're running your code through node, or it's the back end code for a live deployment it'll show the message on the console. If you wanted to show an error to the user you'd need to do something else, have an action triggered by the error. That's where something like a "try... catch" comes in.

"try... catch" is essentially "try and run this function, but if an error is thrown, catch it and do this other thing"

In this function I am essentially using this last throw error as a general fail state. The function first checks to see if the given ID is undefined or not a number, if it passes that check then it runs a search for that ID and if that search doesn't find a student then this error is thrown. If a student is found then the function returns and this line isn't reached.

Make sense?

Comment (procrastian) [reply src/CohortManager.js:137]: Nice one! Thanks for explaining that for me!

**Pull request closed without merging**